### PR TITLE
Fix: attach_device.py: added index option in controller xml

### PR DESCRIPTION
**Pull request by harish-24**
Added index option in controller xml to be attached to the guest. Without index, the test cases fails with, "Cannot parse controller index -1"

Signed-off-by: Harish S <harisrir@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
In virsh.attach_device.controller.type_scsi.normal_test cases

**Before fix:**
2016-12-29 17:28:00,883 process          L0350 INFO | Running '/bin/virsh attach-device virt-tests-vm1-harish /var/tmp/xml_utils_temp_Ihp4Pq.xml  --persistent'
2016-12-29 17:28:00,933 process          L0437 DEBUG| [stderr] error: Failed to attach device from /var/tmp/xml_utils_temp_Ihp4Pq.xml
2016-12-29 17:28:00,934 process          L0437 DEBUG| [stderr] error: internal error: Cannot parse controller index -1
2016-12-29 17:28:00,937 process          L0437 DEBUG| [stdout]
2016-12-29 17:28:00,938 process          L0451 INFO | Command '/bin/virsh attach-device virt-tests-vm1-harish /var/tmp/xml_utils_temp_Ihp4Pq.xml  --persistent' finished with 1 after 0.0510101318359s
2016-12-29 17:28:00,938 virsh            L0691 DEBUG| status: 1
2016-12-29 17:28:00,938 virsh            L0692 DEBUG| stdout:
2016-12-29 17:28:00,939 virsh            L0693 DEBUG| stderr: error: Failed to attach device from /var/tmp/xml_utils_temp_Ihp4Pq.xml
error: internal error: Cannot parse controller index -1
2016-12-29 17:28:00,939 virsh_attach_dev L0350 DEBUG| Attached XML:
2016-12-29 17:28:00,940 virsh_attach_dev L0352 DEBUG| <?xml version='1.0' encoding='UTF-8'?>
2016-12-29 17:28:00,940 virsh_attach_dev L0352 DEBUG| `<controller model="virtio-scsi" type="scsi" />`

**After fix:**
2017-01-09 04:05:14,953 process          L0350 INFO | Running '/bin/virsh attach-device virt-tests-vm1-harish /var/tmp/xml_utils_temp_Tq_duu.xml  --persistent'
2017-01-09 04:05:15,081 process          L0437 DEBUG| [stdout] Device attached successfully
2017-01-09 04:05:15,081 process          L0437 DEBUG| [stdout]
2017-01-09 04:05:15,082 process          L0451 INFO | Command '/bin/virsh attach-device virt-tests-vm1-harish /var/tmp/xml_utils_temp_Tq_duu.xml  --persistent' finished with 0 after 0.125247001648s
2017-01-09 04:05:15,083 virsh            L0691 DEBUG| status: 0
2017-01-09 04:05:15,083 virsh            L0692 DEBUG| stdout: Device attached successfully
2017-01-09 04:05:15,083 virsh            L0693 DEBUG| stderr:
2017-01-09 04:05:15,083 virsh_attach_dev L0350 DEBUG| Attached XML:
2017-01-09 04:05:15,084 virsh_attach_dev L0352 DEBUG| <?xml version='1.0' encoding='UTF-8'?>
2017-01-09 04:05:15,084 virsh_attach_dev L0352 DEBUG| `<controller index="1" model="virtio-scsi" type="scsi" />`

**Comment by will-Do** [comment]
what's your libvirt version?  And hardcode to 1 may not appropriate.
From libvirt.org: http://libvirt.org/formatdomain.html#elementsControllers

> and a mandatory attribute index which is the decimal integer describing in which order the bus controller is encountered (for use in controller attributes of \<address\> elements). Since 1.3.5 the index is optional; if not specified, it will be auto-assigned to be the lowest unused index for the given controller type.

**Comment by harish-24** [comment]
@will-Do Added code for controller index libvirt version < 1.3.5.

**Comment by harish-24** [comment]
@will-Do Can you help in reviewing this?

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_device.py [hunk 1]
```diff
@@ -491,6 +491,15 @@ def init_device(self, index):
         controller_device = controllerclass(type_name=self.type,
                                             virsh_instance=self.test_params.virsh)
         controller_device.model = self.model
+
+        if not libvirt_version.version_compare(1, 3, 5):
+            devices = self.test_params.vmxml.get_devices()
+            controller_devices = devices.by_device_tag('controller')
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_device.py]: can be handled with try and xcepts.LibvirtXMLError

**Comment by harish-24** [comment]
@balamuruhans Thanks for the catch. Updated with changes.

**Pull request merged**